base/files/file_util_starboard.cc

```diff
@@ -219,53 +223,30 @@ bool ReplaceFile(const FilePath& from_path,
   return true;
 }
 
-bool CopyFile(const FilePath &from_path, const FilePath &to_path) {
-  internal::AssertBlockingAllowed();
-
-  base::File source_file(from_path, base::File::FLAG_OPEN | base::File::FLAG_READ);
-  if (!source_file.IsValid()) {
-    DPLOG(ERROR) << "CopyFile(): Unable to open source file: "
-                 << from_path.value();
-    return false;
+#if !BUILDFLAG(IS_APPLE)
+// Mac has its own implementation, this is for all other Posix systems.
+bool CopyFile(const FilePath& from_path, const FilePath& to_path) {
+  ScopedBlockingCall scoped_blocking_call(FROM_HERE, BlockingType::MAY_BLOCK);
+  File infile;
+#if BUILDFLAG(IS_ANDROID)
```

Comment:
why do you need this Android case?

Reply:
This function is borrowed from file_util_posix. I'll cleaned up the non-starboard conditions.